eng/build.ps1

```diff
@@ -369,82 +369,114 @@ function TestUsingRunTests() {
     $env:ROSLYN_TEST_USEDASSEMBLIES = "true"
   }
 
-  $runTests = GetProjectOutputBinary "RunTests.dll" -tfm "net6.0"
+  $testAssemblyFinder = GetProjectOutputBinary "TestAssemblyFinder.dll" -tfm "net6.0"
```

Comment:
the test assembly finder project could probably just be part of the build script, but then I'd have to write it twice, once for powershell and once for bash.  It seemed just complext enough to warrant leaving it in C# so we can re-use it in both.